tests/object_lifetime_tracker_ut/object_lifetime_tracker_ut.c

```diff
@@ -1202,4 +1229,217 @@ TEST_FUNCTION(object_lifetime_tracker_destroy_all_objects_for_key_key_not_found)
+
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    umock_c_reset_all_calls();
+
+    setup_object_lifetime_tracker_act_expectations(test_key_2, test_object_3, 0, 0);
+
+    // act
+    OBJECT_LIFETIME_TRACKER_ACT_RESULT result = object_lifetime_tracker_act(object_lifetime_tracker, test_key_2, test_object_3, test_action_function, test_context);
+
+    // assert
+    ASSERT_ARE_EQUAL(OBJECT_LIFETIME_TRACKER_ACT_RESULT, OBJECT_LIFETIME_TRACKER_ACT_OK, result);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    //cleanup
+    ASSERT_ARE_EQUAL(OBJECT_LIFETIME_TRACKER_UNREGISTER_OBJECT_RESULT, OBJECT_LIFETIME_TRACKER_UNREGISTER_OBJECT_OK, object_lifetime_tracker_unregister_object(object_lifetime_tracker, test_key_2, test_object_3));
+    ASSERT_ARE_EQUAL(OBJECT_LIFETIME_TRACKER_UNREGISTER_OBJECT_RESULT, OBJECT_LIFETIME_TRACKER_UNREGISTER_OBJECT_OK, object_lifetime_tracker_unregister_object(object_lifetime_tracker, test_key_2, test_object_2));
+    ASSERT_ARE_EQUAL(OBJECT_LIFETIME_TRACKER_UNREGISTER_OBJECT_RESULT, OBJECT_LIFETIME_TRACKER_UNREGISTER_OBJECT_OK, object_lifetime_tracker_unregister_object(object_lifetime_tracker, test_key_1, test_object_1));
+    object_lifetime_tracker_destroy(object_lifetime_tracker);
+}
+
+
+/*Tests_SRS_OBJECT_LIFETIME_TRACKER_43_093: [ If there are any failures, object_lifetime_tracker_act shall fail and return OBJECT_LIFETIME_TRACKER_ACT_ERROR. ]*/
+TEST_FUNCTION(object_lifetime_tracker_act_fails_when_underlying_functions_fail)
+{
+    // arrange
+    OBJECT_LIFETIME_TRACKER_HANDLE object_lifetime_tracker = test_create_object_lifetime_tracker();
+    setup_object_lifetime_tracker_register_object_expectations(0, true, 0, true);
+    ASSERT_ARE_EQUAL(OBJECT_LIFETIME_TRACKER_REGISTER_OBJECT_RESULT, OBJECT_LIFETIME_TRACKER_REGISTER_OBJECT_NEW_KEY, object_lifetime_tracker_register_object(object_lifetime_tracker, test_key_1, test_object_1, test_destroy_object, test_destroy_context));
+
+    umock_c_reset_all_calls();
+    umock_c_negative_tests_snapshot();
+
+    for (size_t i = 0; i < umock_c_negative_tests_call_count(); i++)
+    {
+        if (umock_c_negative_tests_can_call_fail(i))
+        {
+            umock_c_negative_tests_reset();
+            umock_c_negative_tests_fail_call(i);
+
+            // act
+            OBJECT_LIFETIME_TRACKER_ACT_RESULT result = object_lifetime_tracker_act(object_lifetime_tracker, test_key_1, test_object_1, test_action_function, test_context);
+
+            // assert
+            ASSERT_ARE_EQUAL(OBJECT_LIFETIME_TRACKER_ACT_RESULT, OBJECT_LIFETIME_TRACKER_ACT_ERROR, result);
+        }
+    }
+
+    // cleanup
+    ASSERT_ARE_EQUAL(OBJECT_LIFETIME_TRACKER_UNREGISTER_OBJECT_RESULT, OBJECT_LIFETIME_TRACKER_UNREGISTER_OBJECT_OK, object_lifetime_tracker_unregister_object(object_lifetime_tracker, test_key_1, test_object_1));
+    object_lifetime_tracker_destroy(object_lifetime_tracker);
+}
```

Comment:
These tests do not yield their maximum potential for coverage. For example, putting a breakpoint in line 490 - it is never hit, therefore line 490 of object_lifetime_tracker.c is not tested.

Other lines where breakpoints are not hit are: 509 and 528.

Reply:
Fixed. I forgot to set up the expectations for the negative test